src/lib.rs

```diff
@@ -1065,14 +1065,71 @@ pub unsafe trait KnownLayout {
     where
         Self: Sized;
 
+    /// `()` for sized types and `usize` for slice DSTs.
+    #[doc(hidden)]
+    type PointerMetadata: PointerMetadata;
```

Comment:
Might want to remove the `doc(hidden)` here once this appears in `where` bounds.

Reply:
Done in the next PR.